include/madara/knowledge/KnowledgeBaseImpl.cpp

```diff
@@ -392,11 +394,11 @@ int KnowledgeBaseImpl::send_modifieds(
 {
   int result = 0;
 
-  MADARA_GUARD_TYPE guard(map_.mutex_);
+  MADARA_GUARD_TYPE guard(transport_mutex_);
 
   if (transports_.size() > 0 && !settings.delay_sending_modifieds)
   {
-    const VariableReferenceMap& modified = map_.get_modifieds();
+    KnowledgeMap modified = map_.get_modifieds_current();
```

Comment:
`reset_modified()` needs to happen atomically along with `get_modifieds_current()`. As-is, there's a race condition, since updates that arrive after this line, but before line 415, will never get sent.

Reply:
addressing